asv/util.py

```diff
@@ -569,8 +569,9 @@ def sig_forward(signum, frame):
         proc.stdout.flush()
         proc.stderr.flush()
 
-        stdout_chunks.append(proc.stdout.read())
-        stderr_chunks.append(proc.stderr.read())
+        # Reading from stdout/stderr hangs non-deterministically
```

Comment:
wait this class reads from stdout and stderr right?

Is this the right fix? where is this used?

Reply:
I'm just not totally sure if this affects anything else downstream

Reply:
This is used to capture output from the benchmarks to later display them to the console.

In addition, it looks like it's used to fill in some default values (e.g. python version when setting up ASV).

Reply:
It's still not totally clear why this is the right thing to do?

It seems like it could affect a lot of ASV functionality which could be problematic as we use this further. Perhaps the right thing to do instead is not to modify this function but rather to add a flag upstream that can be triggered via command line.

Or convince me otherwise?

Reply:
@pcmoritz Looks like asv also calls this function and uses the stdout to set certain default values during setup (e.g. python version). It doesn't look like it's use anywhere critical though.

Reply:
Is this just rewriting https://docs.python.org/3/library/subprocess.html#subprocess.check_output ?

Reply:
Mostly, but there are some modifications (e.g. different kwargs).